### DISCUSSION TITLE
Separating two nodel systems on the same network: Ability to override discovery and advertisement multicast group

### DISCUSSION THREAD

**Discussion by justparking** [museumsvictoria/nodel#338] [General]
This is a proposal to allow the default multicast group to be overridden in the `bootstrap.json`.

The scenario is where you might have two or more nodel systems running on the same network and you don't want either to know about the other.

It might be in a testing environment that shares the same discoverable network (normally VLAN) as the production system.

Or two production systems that you'd like to be completed isolated from each other.

So it would mean all the node listings (especially during bindings) would be separate.

Part of the example **bootstrap.json** file would look like this:

```
{   "nodelHostPort": 8085,
    "discoveryAddress": "224.0.0.252:5354", ⬅ NEW OPTION
    "nodelRoot": "nodes",
    "contentDirectory": ".nodel/webui_cache",
    "cacheDirectory": "cache",
    "networkInterfaces": ["lo", "eth6","eth28"], 
    … }
```

Where `discoveryAddress` takes `udp_multicastgroup:udp_port`.

It's important `udp_port` is specified to ensure there aren't "leaked" packets via direct UDP unicast that could accidentally compromise isolation.

@scroix @dargs @thefennecdeer @paxofsnax - any of you might find this a useful feature?

It's not a difficult one to implement.

**Comment by scroix** [comment]
This seems like a great feature -- especially if the lift is minor.

I’m curious whether this would have implications for simplifying a containerised instance of a Nodehost, specifically regarding port-forwarding the discovery mechanism to the host network.

I tried this many years ago and couldn’t make the host discoverable.
